opnode/rollup/driver/state.go

```diff
@@ -368,31 +376,48 @@ func (s *state) loop() {
 			if err != nil {
 				s.log.Error("Error in handling new L1 Head", "err", err)
 			}
-			// Run step if we are able to
+
+			// The block number of the L1 origin for the L2 safe head is at least SeqWindowSize
+			// behind the L1 head. We can therefore attempt to shift the safe head forward by at
+			// least one L1 block. If the node is holding on to unsafe blocks, this may trigger a
+			// reorg on L2 in the case that safe (published) data conflicts with local unsafe
+			// block data.
 			if s.l1Head.Number-s.l2SafeHead.L1Origin.Number >= s.Config.SeqWindowSize {
 				s.log.Trace("Requesting next step", "l1Head", s.l1Head, "l2Head", s.l2Head, "l1Origin", s.l2SafeHead.L1Origin)
-				requestStep()
+				reqStep()
 			}
-		case <-stepRequest:
+
+		case <-stepReqCh:
 			ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
 			reorg, err := s.handleEpoch(ctx)
 			cancel()
 			if err != nil {
 				s.log.Error("Error in handling epoch", "err", err)
 			}
+
 			if reorg {
 				s.log.Warn("Got reorg")
+
+				// If we're in sequencer mode and experiencing a reorg, we should request a new
+				// block ASAP. Not strictly necessary but means we'll recover from the reorg much
+				// faster than if we waited for the next tick.
 				if s.sequencer {
-					createBlock()
+					reqL2BlockCreation()
 				}
 			}
 
-			// Immediately run next step if we have enough blocks.
+			// The block number of the L1 origin for the L2 safe head is at least SeqWindowSize
+			// behind the L1 head. We can therefore attempt to shift the safe head forward by at
+			// least one L1 block. If the node is holding on to unsafe blocks, this may trigger a
+			// reorg on L2 in the case that safe (published) data conflicts with local unsafe
+			// block data.
```

Comment:
While strictly true, the comment on reorgs is a bit weird to put here, since all this does is recursively cause the `select` we are already in to be executed again.